### PR TITLE
Feature/lfx 1810 project UI

### PR DESCRIPTION
**Pull request by emlimlf**
# Changes proposed ✍️

### What
- Updated the projects group list and the projects list view with new designs
- Implemented "lazy loading" pagination style on both views
- Some minor refactoring to make code more readable

### Why
Requested on [this ticket](https://linear.app/lfx/issue/LFX-1810/new-ui-for-projects-page)

## Checklist ✅
- [ ] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screenshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

## Release Notes

- **New Features**
	- Introduced new components: `LfPill`, `LfStatusPill`, `LfProjectCount`, `AppLfProjectColumn`, `LfProjectIntegrationColumn`, and `LfProjectGroupsDropdown` for enhanced UI elements.
	- Added infinite scrolling pagination to project tables for improved navigation.
	- New `btn--interactive` class for buttons, enhancing user interaction.

- **Improvements**
	- Updated button styles for consistency and visual appeal.
	- Enhanced tab navigation in the admin panel with new tabs for better user experience.
	- Improved project status display with new visual indicators.
	- Refined styling for tabs and buttons to enhance overall aesthetics.
	- Increased spacing and adjusted colors for tab components.
	- Enhanced dropdown menu functionality with conditional rendering based on user permissions.
	- Added hover effects for table rows to improve user interaction.
	- Adjusted table layout and pagination logic for better usability.

- **Bug Fixes**
	- Resolved inconsistencies in button hover states and tab content rendering.

- **Documentation**
	- Added Storybook configurations for the `Pill` component to showcase variations.

### CODE REVIEW

File: frontend/src/assets/scss/tabs.scss [hunk 1]
```diff
@@ -3,22 +3,32 @@
     @apply m-0 border-none #{!important};
   }
   &__active-bar {
-    @apply bg-primary-500 #{!important};
+    @apply bg-transparent #{!important};
   }
```
Comment (joanagmaia) [inline frontend/src/assets/scss/tabs.scss:7]: Hey, this file is still from the old approach.
Gasper has created new components for the new design Nuno has been building inside the ui-kit folder.

If you check `ui-kit/tabs/Tabs.vue`, you can see the tabs tab I believe should be used

Comment (joanagmaia) [reply frontend/src/assets/scss/tabs.scss:7]: Can you also revert the changes on this file? I believe this will update other tabs in the app which is not desirable

File: frontend/src/shared/pill/pill.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<template>
+  <div class="lf-pill" :class="props.color + ' ' + props.type">
+    <slot />
+  </div>
+</template>
```
Comment (joanagmaia) [inline frontend/src/shared/pill/pill.vue:5]: For all re sharable components, Gasper is adding them under ui-kit.
Can you check if there's already any compoennt there that fits this need?
You can run storybook locally (check package.json) to check the available components.
If it doesn't exist, please create this component in the ui-kit folder with the same structure as the remaining components

File: frontend/src/modules/lf/segments/store/state.ts [hunk 3]
```diff
@@ -1,30 +1,32 @@
 import { ProjectGroup, Project } from '@/modules/lf/segments/types/Segments';
 
 export interface SegmentsState {
-  selectedProjectGroup: ProjectGroup | null;
+  selectedProjectGroup: ProjectGroup | null
   adminProjectGroups: {
-    list: ProjectGroups[],
+    list: ProjectGroups[]
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/store/state.ts:6]: I don't think this type exists, might be a typo

**Review by joanagmaia — CHANGES_REQUESTED** [review]
Thanks for the changes Efren.
Overall comment:
- Please check that everything is implemented according to the designs. It seems that a lot of font sizes, spaces, font families are off. Please check Figma and make sure you use the correct variables
- For all components you create that are in shared components, check if they already exist in the ui-kit folder first (run storybook locally might be easier to check). If not then create them there. Instead of creating them on shared/shared-module.js or directly under shared.
- All styling and variables are now set under frontend/config/styles. These are the ones you should use or update if needed. Please don't use or update the ones under src/assets.

Let me know if you need any help going over the codebase to check on some of these files and components structure.

File: frontend/src/assets/scss/tabs.scss (hunk 1 above)
Comment (emlimlf) [reply frontend/src/assets/scss/tabs.scss:7]: I reverted this and used the one in the ui-kit. However, I notice the one in the ui-kit doesn't match the designs. What should be done in the case?

File: frontend/src/shared/pill/pill.vue (hunk 2 above)
Comment (emlimlf) [reply frontend/src/shared/pill/pill.vue]: This doesn't exist there. I moved this component under the ui-kit folder

File: frontend/src/modules/lf/segments/store/state.ts (hunk 3 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/store/state.ts]: yes it was a typo

File: frontend/src/shared/pill/pill.vue (hunk 2 above)
Comment (emlimlf) [reply frontend/src/shared/pill/pill.vue]: I pushed this fix

**Comment by emlimlf** [comment]
@joanagmaia pushed my changes and fixes for the design QA as well.

File: frontend/src/modules/lf/segments/components/view/lf-project-groups-table.vue [hunk 4]
```diff
@@ -1,72 +1,46 @@
...
+      @row-click="handleRowClick"
     >
-      <!-- Name -->
+      <!-- Status -->
       <el-table-column
-        label="Project Group"
-        prop="name"
-        width="300"
-        fixed
+        label="Status"
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/components/view/lf-project-groups-table.vue:13]: The column headers are not matching the designs:
They are with 12px instead of 13px and the spacing around is bigger than it should be.
I think here either we override the element-plus styling for the cells header, or we try to use the table elements from the ui-kit which should offer more flexibility

File: frontend/src/modules/lf/config/status.js [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
+    color: 'yellow',
     label: 'Prospect',
     value: 'prospect',
   },
   {
-    color: 'bg-gray-400',
+    color: 'gray',
     label: 'Archived',
     value: 'archived',
   },
```
Comment (joanagmaia) [inline frontend/src/modules/lf/config/status.js:21]: Not sure if it's related, but we used to have dots with colors before the options on the dropdown, perhaps this change caused that.
I think this `color`'s purpose is to use directly in the template, and not to use it as a class
Screenshot 2024-12-04 at 08 10 38

File: frontend/src/ui-kit/tabs/tabs.scss [hunk 6]
```diff
@@ -1,28 +1,27 @@
-.c-tabs{
-  background: var(--lf-tabs-background);
+.c-tabs {
+  background: var(--lf-color-white);
```
Comment (joanagmaia) [inline frontend/src/ui-kit/tabs/tabs.scss:2]: Could we update `--lf-tabs-background` instead?

File: frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<template>
+  <div class="min-w-[105px]">
+    <lf-pill color="transparent" type="bordered" class="!flex items-center gap-1 w-full">
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue:3]: This is not following the design. On the Status chips/pills, the width is fixed. But I don't think it's the case for this one, might be worth checking with Nuno. Also, "Project" shouldn't be capitalized
Screenshot 2024-12-04 at 08 18 51

File: frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+            <div v-if="progressError" class="text-xs text-gray-500 flex items-center">
+              <lf-icon name="triangle-exclamation" type="light" :size="13" class="text-yellow-600" />
+              Error loading progress
+            </div>
+          </template>
+        </div>
+      </el-popover>
+    </div>
+  </div>
+  <span v-else class="text-gray-400 text-sm">No integrations</span>
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue:85]: Should be "No integrations connected" and with a different style, italic
Screenshot 2024-12-04 at 08 21 54

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue [hunk 9]
```diff
@@ -2,32 +2,37 @@
...
         v-if="pagination.total && hasPermission(LfPermission.projectCreate) && hasAccessToSegmentId(route.params.id)"
-        class="btn btn--md btn--primary"
+        size="medium"
+        type="secondary-ghost"
         @click="onAddProject"
       >
+        <lf-icon name="layer-plus" type="regular" />
         Add project
-      </el-button>
+      </lf-button>
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/pages/lf-projects-page.vue:35]: Next to this button there is an action menu, Probably to edit the project group. Can you check with Nuno?
Screenshot 2024-12-04 at 08 23 39

File: frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+              :color="integration.platform === 'github' && integration?.type === 'mapped' ? 'gray' : 'black'"
+            />
+
+            <lf-icon
+              v-if="integration.status === 'no-data'"
+              name="triangle-exclamation"
+              type="solid"
+              :size="12"
+              class="absolute right-0 top-0 leading-3 text-yellow-500"
+            />
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue:33]: I believe you didn't update the styling of the connected platforms, can you update to match the designs?
Screenshot 2024-12-04 at 08 27 50
Screenshot 2024-12-04 at 08 28 25

File: frontend/src/modules/lf/segments/components/view/lf-projects-table.vue [hunk 11]
```diff
@@ -6,156 +6,48 @@
     row-key="id"
     :resizable="false"
   >
+    <!-- Status -->
+    <el-table-column width="130" class-name="table-columns !bg-white">
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/components/view/lf-projects-table.vue:10]: I think spacing of both the pill and the cells look off. Also not sure about the rounded border of the table. Worth checking
Screenshot 2024-12-04 at 08 30 18
Screenshot 2024-12-04 at 08 30 27

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue [hunk 12]
```diff
@@ -2,32 +2,37 @@
...
-      <h4 class="text-gray-900">
-        Manage projects
-      </h4>
-      <el-button
+    <div class="w-full flex items-center justify-between mb-6 pt-7 border-t border-gray-100">
+      <div class="flex items-center gap-4 whitespace-nowrap">
+        <h4 class="text-gray-900 flex items-center">
+          {{ projectGroupForm.name }}
+        </h4>
+        <app-lf-status-pill :status="projectGroupForm.status" />
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/pages/lf-projects-page.vue:23]: On the designs this one doesn't have a fixed width.
Can you align with Nuno if it's a design inconsistency or not?
If it's on purpose than the component should be more flexible, and we should allow to specify width behaviour

**Review by joanagmaia — CHANGES_REQUESTED** [review]

File: frontend/src/modules/lf/segments/components/view/lf-project-groups-table.vue (hunk 4 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/view/lf-project-groups-table.vue:13]: I'll override the styling for now. The current implementation of the table component in the ui-kit is also not the same as the one in the designs

File: frontend/src/assets/scss/tabs.scss [hunk 13]
```diff
@@ -11,8 +11,9 @@
   &__item {
     @apply font-normal text-gray-900 px-4;
     transition: all 0.2s ease;
+
     &.is-active {
-      @apply font-medium text-primary-900 #{!important};
+      @apply font-medium text-gray-900 #{!important};
```
[the remark this answers is not in the document]
Comment (emlimlf) [reply frontend/src/assets/scss/tabs.scss]: done

File: frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+import { computed } from 'vue';
+import LfPill from '@/ui-kit/pill/Pill.vue';
+import LfIconOld from '@/ui-kit/icon/IconOld.vue';
+
+const props = defineProps<{
+  count: number;
+}>();
+
+const label = computed(() => (props.count > 1 ? 'projects' : 'project'));
+</script>
```
[the remark this answers is not in the document]
Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue]: done

File: frontend/src/modules/lf/segments/components/fragments/lf-status-pill.vue [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+
+<script lang="ts">
+export default {
+  name: 'LfStatusPill',
+};
+</script>
+<style lang="scss">
+.lf-pill {
+  @apply capitalize;
+}
```
[the remark this answers is not in the document]
Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/fragments/lf-status-pill.vue]: done

File: frontend/src/modules/lf/segments/components/view/lf-project-groups-table.vue (hunk 4 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/view/lf-project-groups-table.vue:13]: Pushed the changes

File: frontend/src/modules/lf/config/status.js (hunk 5 above)
Comment (emlimlf) [reply frontend/src/modules/lf/config/status.js]: done

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue (hunk 12 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/pages/lf-projects-page.vue:23]: Are you referring to the status pill? I didn't put a fix width to this.

File: frontend/src/modules/lf/segments/components/view/lf-projects-table.vue (hunk 11 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/view/lf-projects-table.vue:10]: done

File: frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue (hunk 10 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue]: done

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue (hunk 12 above)
Comment (joanagmaia) [reply frontend/src/modules/lf/segments/pages/lf-projects-page.vue:23]: Yeah. I believe it looks good now

File: frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue [hunk 16]
```diff
@@ -0,0 +1,125 @@
+<template>
+  <div
+    v-if="props.integrations?.length"
+    class="flex gap-1 items-center py-2"
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue:4]: Should be gap-2

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue [hunk 17]
```diff
@@ -2,32 +2,46 @@
...
-        Manage projects
-      </h4>
-      <el-button
-        v-if="pagination.total && hasPermission(LfPermission.projectCreate) && hasAccessToSegmentId(route.params.id)"
-        class="btn btn--md btn--primary"
-        @click="onAddProject"
-      >
-        Add project
-      </el-button>
+    <div class="w-full flex items-center justify-between mb-6 pt-7 border-t border-gray-100">
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/pages/lf-projects-page.vue:18]: I don't think the rows should have a border. Can you remove?
Screenshot 2024-12-05 at 12 32 15

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue [hunk 18]
```diff
@@ -2,32 +2,46 @@
...
+      <div class="flex items-center gap-4 whitespace-nowrap">
+        <h4 class="text-gray-900 flex items-center">
+          {{ projectGroupForm.name }}
+        </h4>
+        <app-lf-status-pill :status="projectGroupForm.status" />
+        <app-lf-project-count :count="projectGroupForm.projects?.length" />
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/pages/lf-projects-page.vue:24]: This value is not being updated whenever I create a new project in the project group. I need to refresh to see it

**Review by joanagmaia — COMMENTED** [review]
It's looking great 👌 Just added a couple more comments

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue (hunk 18 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/pages/lf-projects-page.vue]: done

File: frontend/src/modules/lf/segments/pages/lf-projects-page.vue (hunk 17 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/pages/lf-projects-page.vue:18]: done, you've highlighted a different line of code but it's fine I've fixed it.

File: frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue (hunk 16 above)
Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/fragments/lf-project-integration-column.vue]: done

File: frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue [hunk 19]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <div class="min-w-[105px]">
```
Comment (joanagmaia) [inline frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue:2]: Why is this needed? I think this is what was causing the fixed width I referred previously.
Can you remove it?

Current implementation:
Screenshot 2024-12-05 at 18 36 39

Screenshot 2024-12-05 at 18 36 25

Design:
Screenshot 2024-12-05 at 18 37 03

**Review by joanagmaia — APPROVED** [review]
Looks good to me.
Can you just check if it's ok to remove the class I'm referring to in my last comment? It's causing a fixed width on one of the pill components, and I don't think it's supposed to happen

File: frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue (hunk 19 above)
Comment (joanagmaia) [reply frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue:2]: Hey @emlimlf did you check my last comment?
Let me know once you are able to check this one out, and I'll merge the PR after

Comment (emlimlf) [reply frontend/src/modules/lf/segments/components/fragments/lf-project-count.vue]: @joanagmaia sorry I must've missed this. I pushed the change
